services/net/extract-quotes/ExtractQuotesManager.cs

```diff
@@ -0,0 +1,287 @@
+        {
+            if (consume.Error.IsFatal)
+                this.Listener.Stop();
+        }
+    }
+
+    /// <summary>
+    /// The Kafka consumer has stopped which means we need to also cancel the background task associated with it.
+    /// </summary>
+    /// <param name="sender"></param>
+    /// <param name="e"></param>
+    private void ListenerStopHandler(object sender, EventArgs e)
+    {
+        if (_consumer != null &&
+            !_notRunning.Contains(_consumer.Status) &&
+            _cancelToken != null && !_cancelToken.IsCancellationRequested)
+        {
+            _cancelToken.Cancel();
+        }
+    }
+
+    /// <summary>
+    /// Retrieve a file from storage and send to Microsoft Cognitive Services. Obtain
+    /// the report and update the content record accordingly.
+    /// </summary>
+    /// <param name="result"></param>
+    /// <returns></returns>
+    private async Task HandleMessageAsync(ConsumeResult<string, IndexRequestModel> result)
+    {
+        try
+        {
+            // The service has stopped, so to should consuming messages.
+            if (this.State.Status != ServiceStatus.Running)
+            {
+                this.Listener.Stop();
+                this.State.Stop();
+            }
+            else
+            {
+                await ProcessIndexRequestAsync(result);
+
+                // Inform Kafka this message is completed.
+                this.Listener.Commit(result);
+                this.Listener.Resume();
+
+                // Successful run clears any errors.
+                this.State.ResetFailures();
+                _retries = 0;
+            }
+        }
+        catch (Exception ex)
+        {
+            if (ex is HttpClientRequestException httpEx)
+            {
+                this.Logger.LogError(ex, "HTTP exception while consuming. {response}", httpEx.Data["body"] ?? "");
+                await this.SendEmailAsync("HTTP exception while consuming. {response}", ex);
+            }
+            else
+            {
+                this.Logger.LogError(ex, "Failed to handle message");
+                await this.SendEmailAsync("Failed to handle message", ex);
+            }
+            ListenerErrorHandler(this, new ErrorEventArgs(ex));
+        }
+        finally
+        {
+            if (State.Status == ServiceStatus.Running) Listener.Resume();
+        }
+    }
+
+    /// <summary>
+    /// Process the report request.
+    /// </summary>
+    /// <param name="request"></param>
+    /// <returns></returns>
+    private async Task ProcessIndexRequestAsync(ConsumeResult<string, IndexRequestModel> result)
+    {
+        this.Logger.LogDebug($"ProcessIndexRequestAsync:BEGIN:{result.Message.Key}");
+
+        this.Logger.LogInformation("Indexing content from Topic: {Topic}, Content ID: {Key}", result.Topic, result.Message.Key);
+        var model = result.Message.Value;
+
+        if (model.Action == IndexAction.Index || model.Action == IndexAction.Publish)
```

Comment:
Is this logic relevant?

Reply:
I thought it would be.  The story says that the extract quotes should listen to the index topic.  Based on that I figured we would only attempt to extract quotes from content if the action was index or publish

Reply:
One thing to be aware of is that your extracted quotes won't show up in Elasticsearch.  Not sure if they need to be.

Reply:
yah I kind of talked about that in one of my other comments.  Since the quote came from the content is there a need to have a seperate index of quotes?  The business might say, I want to find a quote from an individual that contains this.  If we dont use elastic for that kind of index, it wont be as *nice*